src/document.rs

```diff
@@ -575,37 +565,79 @@ impl Document {
     }
 }
 
-pub struct Entry<'a> {
-    inner: linked_hash_map::Entry<'a, String, Bson>,
+/// A view into a single entry in a map, which may either be vacant or occupied.
+///
+/// This enum is constructed from the entry method on HashMap.
+pub enum Entry<'a> {
```

Comment:
I updated this to match both `IndexMap` and `std::HashMap`'s `Entry` API. I updated just the parts that are breaking and left the remaining bits to future, additive work. I'll file a ticket once this is merged for us to remember to do the rest of it. It's basically just adding a few methods to `OccupiedEntry` and `VacantEntry`.

Reply:
Yep, `OccupiedEntry` will have a `fn get() -> &Bson` on it which will return the value.

In the old API (and the new one, for now), it looks like you had to use `or_insert` or `or_insert_with` to get the value out from an entry, inserting into the into it if it wasn't occupied. It was definitely incomplete and missing a lot of the useful parts of the `std::HashMap` one. 

The `Entry` API looks mostly useful for mutating / inspecting the map itself without having to compute the hash of an input multiple times. I remember we were doing that in the driver and a user actually submitted a PR to update it to use `entry` and be a little more efficient.
```rust
let mut map = HashMap::new();
// old
if map.get("key").is_none() {
    map.insert("key".to_string(), "value".to_string());
}

// new
if let Entry::Vacant(entry) = map.entry("key") {
    entry.insert("value".to_string());
}
```

Reply:
ahh I see. thanks for explanation! cool, seems like the new API should be a nice usability improvement then.